### PR TITLE
Enable NRT for multiplayer and playlists

### PR DESCRIPTION
**Pull request by smoogipoo**
Most relevant changes:
- `RoomSubScreen.CreateGameplayScreen()` - `PlaylistItem` parameter added with non-null precondition in `RoomSubScreen.StartPlay()`.
- `DrawableRoomPlaylistItem` - this class is totally boned, with a virtual method call that's never called in the base class. Fields made nullable because there is at least one usage that doesn't call said virtual method at all.
- `OnlinePlayScreen` - moved screen stack construction to class init because it's referenced externally via `CursorVisible`.

### CODE REVIEW

**Comment by bdach** [comment]
> `DrawableRoomPlaylistItem` - this class is totally boned, with a virtual method call that's never called in the base class. Fields made nullable because there is at least one usage that doesn't call said virtual method at all.

Yeah that one is a stinker. To call it overloaded with functionality is underselling it. If you find yourself taking a hammer to that atrocity I'd be overjoyed.

**Comment by smoogipoo** [comment]
It'll happen at some point for sure...

**Review by bdach — COMMENTED** [review]
Seems generally ok

File: osu.Game/Screens/OnlinePlay/Match/Components/MatchTypePicker.cs [hunk 1]
```diff
@@ -84,7 +83,7 @@ public GameTypePickerItem(MatchType value)
                 };
             }
 
-            private Sample selectSample;
+            private Sample selectSample = null!;
```
[the remark this answers is not in the document]
Comment (smoogipoo) [reply osu.Game/Screens/OnlinePlay/Match/Components/MatchTypePicker.cs:86]: Yeah this is a very weird one and I was a bit conflicted in what changes to make. I decided to keep existing code as much as possible at the end of the day such that if something was used in a may-be-null way then I tended towards assuming null is still a valid value, and suspending my own analysis. Except things in `load()` and similar.

We can probably do a pass through the rest of the game and remove this pattern later.

**Review by bdach — APPROVED** [review]
not gonna overthink this

**Pull request merged**